### PR TITLE
add optional of start bypass

### LINKED ISSUE #294: Bypass controller should not be coupled ads or worklaod mode
**Task description**:

Curretly bypass controller only start when kmesh works in ads or workload mode.  https://github.com/kmesh-net/kmesh/blob/ac76f79373da36fec4cbac50cb150c2cbd96cfdd/pkg/controller/controller.go#L48

But as i can see bypass can be used separately. So we may need to allow it running alone

**Solution**:

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**

**What this PR does / why we need it**:
add optional of start bypass, default is false
**Which issue(s) this PR fixes**:
Fixes #294 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -34,20 +34,23 @@ var (
 )
 
 func Start() error {
-	if !bpfConfig.AdsEnabled() && !bpfConfig.WdsEnabled() {
-		return fmt.Errorf("controller start failed")
-	}
+	if bpfConfig.EnableByPass {
+		client = NewXdsClient()
```
Comment (weli-l) [inline pkg/controller/controller.go:38]: client should be declared outside bypass

Comment (LiZhenCheng9527) [reply pkg/controller/controller.go]: Done

File: daemon/options/bpf.go [hunk 2]
```diff
@@ -39,6 +40,7 @@ func (c *BpfConfig) AttachFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&c.Cgroup2Path, "cgroup2-path", "/mnt/kmesh_cgroup2", "cgroup2 path")
 	cmd.PersistentFlags().StringVar(&c.Mode, "mode", "workload", "controller plane mode, valid values are [ads, workload]")
 	cmd.PersistentFlags().BoolVar(&c.EnableMda, "enable-mda", false, "enable mda")
+	cmd.PersistentFlags().BoolVar(&c.EnableByPass, "bypass", false, "optional of start bypass, default is false")
```
Comment (hzxuzhonghu) [inline daemon/options/bpf.go:43]:
```suggestion
	cmd.PersistentFlags().BoolVar(&c.EnableByPass, "enable-bypass", false, "whether to start bypass controller or not, default to false")
```

Comment (hzxuzhonghu) [inline daemon/options/bpf.go:43]: BTW, i donot think bypass belong to bpf. We can introduce a controller options

File: deploy/helm/values.yaml [hunk 3]
```diff
@@ -9,6 +9,7 @@ deploy:
     imagePullPolicy: IfNotPresent
     containers:
       kmeshDaemonArgs: "--mode=workload"
+      kmeshByPass: "--bypass=false"
```
[the remark this answers is not in the document]
Comment (nlgwcy) [reply deploy/helm/values.yaml:12]: maybe no need to add extra parameters. If bypass tag is configured for the pod managed by Kmesh, the bypass tag takes effect.

File: daemon/options/bypass.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package options
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type byPassConfig struct {
+	EnalbeByPass bool
```
Comment (hzxuzhonghu) [inline daemon/options/bypass.go:23]: typo

File: deploy/helm/values.yaml [hunk 5]
```diff
@@ -5,11 +5,10 @@ deploy:
       xdsAddress: istiod.istio-system.svc:15012
     image:
       repository: ghcr.io/kmesh-net/kmesh
-      tag: latest
+      tag: 5.13.2
```
Comment (hzxuzhonghu) [inline deploy/helm/values.yaml:8]: revert

File: pkg/controller/controller.go [hunk 6]
```diff
@@ -35,24 +35,30 @@ type Controller struct {
...
+
+		err = bypass.StartByPassController(clientset)
+		if err != nil {
+			return fmt.Errorf("failed to start bypass controller: %v", err)
+		}
 
-	err = bypass.StartByPassController(clientset)
-	if err != nil {
-		return fmt.Errorf("failed to start bypass controller: %v", err)
+		log.Info("enable bypass successful")
```
Comment (hzxuzhonghu) [inline pkg/controller/controller.go:61]: nit: start bypass controller sucessfully

**Review by hzxuzhonghu — APPROVED** [review]
@weli-l  can you also take a look

**Pull request merged**